File: hamersapp/src/main/java/nl/ecci/hamers/news/NewNewsActivity.java
package nl.ecci.hamers.news;

import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.view.View;
import android.widget.EditText;

import com.android.volley.VolleyError;

import org.json.JSONException;
import org.json.JSONObject;

import nl.ecci.hamers.MainActivity;
import nl.ecci.hamers.R;
import nl.ecci.hamers.helpers.HamersActivity;
import nl.ecci.hamers.helpers.NewItemActivity;
import nl.ecci.hamers.loader.Loader;
import nl.ecci.hamers.loader.PostCallback;

public class NewNewsActivity extends NewItemActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.news_new_acitivity);

        final android.support.v7.widget.Toolbar toolbar = (android.support.v7.widget.Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        final ActionBar actionBar = getSupportActionBar();

        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setHomeButtonEnabled(true);
        }
    }

    public void postItem() {
        EditText news_title = (EditText) findViewById(R.id.news_title);
        EditText news_body = (EditText) findViewById(R.id.news_body);

        String title = news_title.getText().toString();
        String newsBody = news_body.getText().toString();

        JSONObject body = new JSONObject();
        try {
            body.put("title", title);
            body.put("body", newsBody);
            body.put("cat", "l");
        } catch (JSONException ignored) {
        }

        Loader.postOrPatchData(Loader.NEWSURL, body, -1, this, MainActivity.prefs, new PostCallback() {
            @Override
            public void onSuccess(JSONObject response) {
                finish();
            }

            @Override
            public void onError(VolleyError error) {

            }
        });
    }
}

File: hamersapp/src/main/java/nl/ecci/hamers/news/NewNewsActivity.kt
package nl.ecci.hamers.news

import android.os.Bundle
import com.android.volley.VolleyError
import kotlinx.android.synthetic.main.news_new_acitivity.*
import nl.ecci.hamers.R
import nl.ecci.hamers.helpers.NewItemActivity
import nl.ecci.hamers.loader.Loader
import nl.ecci.hamers.loader.PostCallback
import org.json.JSONException
import org.json.JSONObject

class NewNewsActivity : NewItemActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.news_new_acitivity)

        initToolbar()
    }

    override fun postItem() {
        val body = JSONObject()
        try {
            body.put("name", news_title.text.toString())
            body.put("body", news_body.text.toString())
            body.put("cat", "l")
        } catch (ignored: JSONException) {
        }

        Loader.postOrPatchData(this, Loader.NEWSURL, body, -1, object : PostCallback {
            override fun onSuccess(response: JSONObject) {
                finish()
            }

            override fun onError(error: VolleyError) {}
        })
    }
}